### PR TITLE
[WIP] Update templates for Identity individual auth

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/Identity/Controllers/ApplicationAuthorizationController.cs [hunk 1]
```diff
@@ -61,7 +57,9 @@ public class IdentityServiceController : Controller
                 authorizationResult.User,
                 authorizationResult.Application);
 
-            AddAmbientClaims(context);
+            context.AmbientClaims.Add(new Claim("policy", "signinsignup"));
+            context.AmbientClaims.Add(new Claim("version", "1.0"));
+            context.AmbientClaims.Add(new Claim("tenantId", "CDF07358 -BA97-470F-93CD-FC46E1B57F99"));
```
Comment (mlorbetske) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/Identity/Controllers/ApplicationAuthorizationController.cs]: Is this guid supposed to have a space in it?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/Identity/Controllers/ApplicationAuthorizationController.cs [hunk 2]
```diff
@@ -85,14 +83,16 @@ public class IdentityServiceController : Controller
 
             var context = request.CreateTokenGeneratingContext(session.User, session.Application);
 
-            AddAmbientClaims(context);
+            context.AmbientClaims.Add(new Claim("policy", "signinsignup"));
+            context.AmbientClaims.Add(new Claim("version", "1.0"));
+            context.AmbientClaims.Add(new Claim("tenantId", "CDF07358 -BA97-470F-93CD-FC46E1B57F99"));
```
Comment (mlorbetske) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/Identity/Controllers/ApplicationAuthorizationController.cs]: Same

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs [hunk 3]
```diff
@@ -47,14 +55,28 @@ public Startup(IConfiguration configuration)
...
+                .AddDefaultTokenProviders()
+                .AddApplications()
+                .AddEntityFrameworkStores<IdentityServiceDbContext>()
+                .AddClientExtensions();
+
 #elseif (IndividualB2CAuth)
             services.AddAzureAdB2CAuthentication();
 #elseif (OrganizationalAuth)
             services.AddAzureAdAuthentication();
 #endif
```
Comment (seancpeters) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs]: Do we want to use the commented out conditional syntax for some of this, so the template source would be runnable? (same would apply to other places)

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/appsettings.json [hunk 4]
```diff
@@ -2,33 +2,18 @@
...
-//            },
-//            "LocalhostHttps": {
-//              "Address": "127.0.0.1",
-//              "Port": "43434",
-//              "Certificate": "HTTPS"
-//            }
-//          }
-//        }
-//      },
 //#if (IndividualLocalAuth)
```
Comment (seancpeters) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/appsettings.json]: This file has unbalanced #if & #endif statements. There are 6 #if's, but only 5 #endif's. It appears the old version in this PR has the same problem. I believe the missing #endif should be at approximately line 57 (fine-tune to get the curly brace inclusion-exclusion correct).

Comment (mlorbetske) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/appsettings.json]: The issue with the balancing was fixed with #887, we should make sure that that change gets absorbed in this PR

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs [hunk 5]
```diff
@@ -47,14 +55,28 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
 #if (IndividualLocalAuth)
-            services.AddIdentityServiceAuthentication();
+            services.AddDbContext<IdentityServiceDbContext>(options =>
+                options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
+
+            services.AddIdentity<ApplicationUser, IdentityRole>()
+                .AddDefaultTokenProviders()
+                .AddApplications()
```
Comment (mlorbetske) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs]: I get a compile error with this line indicating that it needs its type parameters specified and a `Action<IdentityServiceOptions>` to be passed in

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs [hunk 6]
```diff
@@ -67,21 +89,22 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
                 app.UseDeveloperExceptionPage();
 #if (IndividualLocalAuth)
                 app.UseDatabaseErrorPage();
-                app.UseDevelopmentCertificateErrorPage(Configuration);
+                app.UseDevelopmentCertificateErrorPage();
```
Comment (mlorbetske) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs]: I get a compile error with the `Configuration` argument removed

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs [hunk 7]
```diff
@@ -47,14 +55,28 @@ public Startup(IConfiguration configuration)
...
         {
 #if (IndividualLocalAuth)
-            services.AddIdentityServiceAuthentication();
+            services.AddDbContext<IdentityServiceDbContext>(options =>
+                options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
+
+            services.AddIdentity<ApplicationUser, IdentityRole>()
+                .AddDefaultTokenProviders()
+                .AddApplications()
+                .AddEntityFrameworkStores<IdentityServiceDbContext>()
```
Comment (mlorbetske) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Startup.cs]: I get a nullref on this line with this stack

```
>	Microsoft.AspNetCore.Identity.Service.EntityFrameworkCore.dll!Microsoft.AspNetCore.Identity.Service.EntityFrameworkCore.IdentityServiceBuilderExtensions.AddEntityFrameworkStores<tmp99.Identity.Data.IdentityServiceDbContext>(Microsoft.AspNetCore.Identity.Service.IIdentityServiceBuilder builder)	Unknown
 	tmp99.dll!tmp99.Startup.ConfigureServices(Microsoft.Extensions.DependencyInjection.IServiceCollection services) Line 41	C#
```

**Comment by mlorbetske** [comment]
Just tried these out with 2.0.0-preview2-25368 packages & added a few more comments to the review

**Comment by javiercn** [comment]
There are a couple of PRs waiting to go in before this one can go in.

**Pull request closed without merging**